docs/getting-started.md

```diff
@@ -1,372 +1,282 @@
 
-Within the Firecracker repository root directory:
+# Set user read permission on the ssh key
+chmod 400 ./ubuntu-18.04.id_rsa
 
-1. With the __default__ musl target:
+# Clone the firecracker repository
+git clone https://github.com/firecracker-microvm/firecracker
 
-   ```tools/devtool build```
+# Start docker
+sudo systemctl start docker
 
-1. (__Experimental only__) using the gnu target:
+# Build firecracker
+#
+# It is possible to build for gnu, by passing the arguments '-l gnu'.
+#
+# This will produce the firecracker and jailer binaries under
+# `./firecracker/build/cargo_target/${toolchain}/debug`.
+#
+sudo ./firecracker/tools/devtool build
 
-   ```tools/devtool build -l gnu```
+API_SOCKET="./firecracker.socket"
 
-This will build and place the two Firecracker binaries at:
+# Remove API unix socket
+rm -f $API_SOCKET
 
-- `build/cargo_target/${toolchain}/debug/firecracker` and
-- `build/cargo_target/${toolchain}/debug/jailer`.
+# Run firecracker
+./firecracker/build/cargo_target/${ARCH}-unknown-linux-musl/debug/firecracker \
+    --api-sock "${API_SOCKET}"
+```
 
-If you would like to test a new feature and work with
-dependencies on libraries located in private git repos, you
-can use the `--ssh-keys` flag to specify the paths to your
-public and private SSH keys on the host. Both of them are
-required for git authentication when fetching the repositories.
+In a new terminal (do not close the 1st one):
 
 ```bash
-tools/devtool build --ssh-keys ~/.ssh/id_rsa.pub ~/.ssh/id_rsa
+TAP_DEV="tap0"
+TAP_IP="172.16.0.1"
+MASK_SHORT="/30"
+
+# Setup network interface
+sudo ip link del "$TAP_DEV" 2> /dev/null || true
+sudo ip tuntap add dev "$TAP_DEV" mode tap
+sudo ip addr add "${TAP_IP}${MASK_SHORT}" dev "$TAP_DEV"
+sudo ip link set dev "$TAP_DEV" up
+
+# Setup ip forwarding
+sudo sh -c "echo 1 > /proc/sys/net/ipv4/ip_forward"
+sudo iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE || true
+sudo iptables -D FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT \
+    || true
+sudo iptables -D FORWARD -i tap0 -o eth0 -j ACCEPT || true
+
+# Enable microVM internert access
+sudo iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+sudo iptables -I FORWARD 1 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
+sudo iptables -I FORWARD 1 -i tap0 -o eth0 -j ACCEPT
+
+API_SOCKET="./firecracker.socket"
+LOGFILE="./firecracker.log"
+
+# Create log file
+touch $LOGFILE
+
+# Set log file
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"log_path\": \"${LOGFILE}\",
+        \"level\": \"Debug\",
+        \"show_level\": true,
+        \"show_log_origin\": true
+    }" \
+    "http://localhost/logger"
+
+# Set machine config
+#
+# Without this the microVM will have the default values of 1 vCPU and 128MB of RAM.
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"vcpu_count\": 2,
+        \"mem_size_mib\": 128,
+        \"cpu_template\": \"C3\"
```

Comment:
do we want our users to jump straight into a template?

Reply:
Good point, do you think this should just be removed?

Reply:
I think it's an advanced feature, I wouldn't encourage using it by default.

Reply:
Okay, I've removed this API call.